torchdata/nodes/prefetch.py

```diff
@@ -15,31 +15,28 @@
 
 class Prefetcher(BaseNode[T]):
     def __init__(self, source: BaseNode[T], prefetch_factor: int, snapshot_frequency: int = 1):
+        super().__init__()
         self.source = source
         self.prefetch_factor = prefetch_factor
         self.snapshot_frequency = snapshot_frequency
         self._it: Optional[_SingleThreadedMapper[T]] = None
         self._iter_for_state_dict: bool = False
 
-    def _get_iterator(self, initial_state: Optional[Dict[str, Any]]) -> _SingleThreadedMapper[T]:
-        return _SingleThreadedMapper(
+    def reset(self, initial_state: Optional[Dict[str, Any]] = None):
+        super().reset(initial_state)
+        if self._it is not None:
+            self._it._shutdown()
+            del self._it
+        self._it = _SingleThreadedMapper(
```

Comment:
TODO: we should try and make single threaded mapper and ParallelMapper handle reset, similar to how `persistent_workers` currently works. However that's going to take a more significant refactor so leaving like this for now to minimize the changes